cpp/docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,453 @@
+The source file is then published under a separate the _Examples_ page in the output.
+
+## Namespaces
+
+Doxygen output includes a _Namespaces_ page that shows all the namespaces declared with comment blocks in the processed files.
+Here is an example of doxygen description comment for a namespace declaration.
+
+```c++
+//! cuDF interfaces
+namespace cudf {
+```
+
+A description comment should be included for only once for each unique namespace declaration.
+Otherwise, if more than one description is found, doxygen will aggregate the descriptions in an arbitrary order in the output pages.
+
+If you are introducing a new namespace, provide a short description block for only one declaration and not for every occurrence.
+
+## Groups/Modules
+
+Grouping declarations into modules helps users to find APIs in the doxygen pages.
+Generally, common functions are already grouped logically into header files but not easily reflected in the doxygen output.
+Doxygen output includes a _Modules_ page that organizes items into groups using the [Grouping doxygen commands](http://www.doxygen.nl/manual/grouping.html)
+These commands can group common functions across header files, source files, and even namespaces.
+Groups can also be nested by defining new groups within existing groups.
+
+For libcudf, all the group hierarchy is defined in the `doxygen_groups.h` header file.
+This file does not need to be included in any other source file.
+It is processed only by the doxygen tool.
+Use this file to add or update any groups.
+Doxygen will also use the definitions in this file to order to groups in the _Modules_ page.
+
+When creating a new API, specify the group in which it should belong using the [@ingroup](http://www.doxygen.nl/manual/commands.html#cmdingroup) tag along with the group reference id from the `doxygen_groups.h` file.
+
+```c++
+namespace cudf {
+
+/**
+ * @brief ...
+ *
+ * @ingroup transformation_fill
+ *
+ * @param ...
+ * @return ...
+ */
+std::unique_ptr<column> fill(table_view const& input,...);
+
+}  // namespace cudf
+```
+
+The `@{` is a convenience to include all doxygen documentation from here to the `@}` line as part of the group.
+
+```c++
+namespace cudf {
+/**
+ * @ingroup transformation_fill
+ * @{
+ */
+
+/**
+ * @brief ...
+ *
+ * @ingroup transformation_fill
+ *
+ * @param ...
+ * @return ...
+ */
+std::unique_ptr<column> fill(table_view const& input,...);
+
+/** @} */
+}  // namespace cudf
+```
+
+This just saves adding `@ingroup` to each comment block that appears in the file.
+Make sure there is a blank line after the group definition comment block so doxygen know it does not apply to whatever follows in the source code.
+Note that the `@{` and `@}` does not work especially if it crosses namespaces or includes classes or enums.
+
+## Build Doxygen Output
+
+The doxygen tool can be installed using the instructions on its [Installation](http://www.doxygen.nl/manual/install.html) page.
```

Comment:
These instructions are for building from source. I would use conda or apt to install

```suggestion

We recommend installing Doxygen using conda (`conda install doxygen`) or a Linux package manager (`sudo apt install doxygen`). Alternatively you can [build and install from source](http://www.doxygen.nl/manual/install.html).
```